File: src/main/java/com/autosale/shop/repository/UserRepository.java
package com.autosale.shop.repository;

import com.autosale.shop.model.User;

import java.util.List;
import java.util.Optional;

public interface UserRepository {

    List<User> findAll();

    Optional<User> findById(int id);

    Optional<User> findByUsername(String userName);

    Optional<Integer> save(User user);

    void update(User user);

    int deleteById(int id);
}

File: src/main/java/com/autosale/shop/repository/UserRepository.kt
package com.autosale.shop.repository

import com.autosale.shop.model.User
import java.util.*

interface UserRepository {
    fun findAll(): List<User>
    fun findById(id: Int): Optional<User>
    fun findByUsername(userName: String): Optional<User>
    fun save(user: User): Optional<Int>
    fun update(user: User)
    fun deleteById(id: Int): Int
}
